dagmc_stats/DagmcFile.py

```diff
@@ -106,6 +109,34 @@ def __set_entityset_ranges(self):
                 self._my_moab_core.get_entities_by_type_and_tag(self.root_set, types.MBENTITYSET,
                                                                 self.dagmc_tags['geom_dim'], [dimension])
 
+    def get_meshset_by_id(self, dim, ids=[]):
+        """Get meshset of the geometry with specified dimension and ids
+
+        inputs
+        ------
+        dim : dimension of the meshset
+        ids : ids of the meshset
+
+        outputs
+        -------
+        meshset : meshset of the geometry with given dimension and ids
+        """
+        # if no id is passed in
+        if len(ids) == 0:
+            return self.entityset_ranges[dim]
+
+        all_global_id = []
+        for id in ids:
+            all_global_id.extend(list(self._my_moab_core.get_entities_by_type_and_tag(self.root_set,
```

Comment:
This (py)MOAB method allows you to specify multiple tags and a value for each of them.  Therefore, you could pass `[dimTag, global_id_tag]` with values `[dim, id]` (or something like that...)

Reply:
OHH I didn't realize that! That's clever.

Reply:
Can you do this for a list of `id`s though? That was part of the reason we have it in a for-loop.

Reply:
The right syntax may actually be:
`[dimTag, global_id_tag]` with values `[ [dim], [id1, id2, id2] ]`
Might be worth a little digging/experimenting.

BTW, I figured this out because this is how we implement the same method in MOAB's GeomTopoTool for the same functionality

Reply:
try this: `meshset = self._my_moab_core.get_entities_by_type_and_tag(self.root_set, types.MBENTITYSET [self.dagmc_tags['geom_dim'],self.dagmc_tags['global_id']], [[dim], ids])`

I think `dim` has to be in brackets

Reply:
@kkiesling I tried this but it still reports error `AssertionError: Only single-tag queries are currently supported.`

Reply:
I think this error message is probably well-formed to tell us that it won't work.  Specifically, it looks like a message @pshriwise wrote to say that this functionality is defined in C++ but not in python... sorry for the misdirection

Reply:
I think a better way to do this is to make an entityset for each dimension that contains all of the entitysets of that dimension, and then pass that as the first entry of `get_by_type_and_tag()`